apache-maven/src/assembly/shared/init

```diff
@@ -50,14 +50,15 @@ if [ -z "$JAVACMD" ] ; then
 fi
 
 if [ ! -x "$JAVACMD" ] ; then
-  echo "The JAVA_HOME environment variable is not defined correctly" >&2
-  echo "This environment variable is needed to run this program" >&2
-  echo "NB: JAVA_HOME should point to a JDK not a JRE" >&2
+  echo "The JAVA_HOME environment variable is not defined correctly," >&2
+  echo "this environment variable is needed to run this program." >&2
   exit 1
 fi
 
 if [ -z "$JAVA_HOME" ] ; then
   echo "Warning: JAVA_HOME environment variable is not set." >&2
+elif [ ! -x "$JAVA_HOME/bin/javac" ] ; then
```

Comment:
Now that I look at this I see a problem. This is testing that JAVA_HOME does not point to a JDK. It is not testing that it does point to a JRE. The problem I encountered when I filed the bug was that JAVA_HOME pointed to a directory that did not exist. Maybe test also that $JAVA_HOME/bin/java does exist?

Reply:
Isn't it done with `if [ ! -x "$JAVACMD" ] ; then`?

Reply:
Honestly not sure. Bash is not my strong suit.

Reply:
Technically, Robert's right on JIRA. Maven does not require a JDK, plugins do.